services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +49,64 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  // Check if the event pointer is valid
+  if (event == NULL) {
+      return ERR_CODE_INVALID_ARG;  // Return error for invalid argument
+  }
 
-  return ERR_CODE_SUCCESS;
+  // Attempt to send the event to the thermal manager queue
+  if (xQueueSend(thermalMgrQueueHandle, event, portMAX_DELAY) == pdPASS) {
+      return ERR_CODE_SUCCESS;  // Return success if the event was sent
+  }
+
+  // Return error if sending the event fails
+  return ERR_CODE_QUEUE_FULL;  // Return specific error code for queue full
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+
+  // Variable to store the current temperature
+  float temperature;
+
+  // Read the current temperature from the LM75BD sensor
+  error_code_t result = readTempLM75BD(config->devAddr, &temperature); // Call with pointer
```

Comment:
we do not want to be calling driver functions from within an ISR. think about other potential places we can read temp and check for overtemperature.